flink-queryable-state/flink-queryable-state-client-java/src/main/java/org/apache/flink/queryablestate/network/Client.java

```diff
@@ -166,27 +168,39 @@ public String getClientName() {
 	 *
 	 * <p>After a call to this method, all returned futures will be failed.
 	 */
-	public void shutdown() {
+	public CompletableFuture<Boolean> shutdown() {
```

Comment:
This method when being called twice will give you a wrong result. The second call will immediately return you a completed future.